lnwallet/channel.go

```diff
@@ -6019,26 +6034,50 @@ func (lc *LightningChannel) availableCommitmentBalance(view *htlcView) (
 		htlcTimeoutFee(feePerKw),
 	)
 
+	// On the remote commitment, the HTLC success transaction will be used.
+	if remoteChain {
+		htlcFee = lnwire.NewMSatFromSatoshis(
+			htlcSuccessFee(feePerKw),
+		)
+	}
+
 	// The HTLC output will be manifested on the commitment if it
 	// is non-dust after paying the HTLC fee.
 	nonDustHtlcAmt := dustlimit + htlcFee
 
+	switch {
+
 	// If we are the channel initiator, we must to subtract the commitment
 	// fee from our available balance.
-	if lc.channelState.IsInitiator {
+	case lc.channelState.IsInitiator:
 		switch {
 		// If our balance is high enough to pay for a non-dust HTLC and
 		// the resulting commitment fee, subtract this commitment fee
 		// to reflect what we have left for such payments.
 		case ourBalance >= htlcCommitFee+nonDustHtlcAmt:
 			ourBalance -= htlcCommitFee
 
+		// We have enough to pay the non-dust HTLC, but not enough to
+		// cover the resulting fee. Therefore we set our available
+		// balance slightly below the non-dust amount to allow HTLCs up
+		// to that size.
+		case ourBalance >= baseCommitFee+nonDustHtlcAmt:
+			ourBalance = nonDustHtlcAmt - 1
+
 		// Otherwise, what's left after subtracting the base commitment
 		// fee will be less than the non-dust HTLC amount, and that's
 		// what we have available as a balance.
 		default:
 			ourBalance -= baseCommitFee
 		}
+
+	// If we're not the initiator, we must check whether the remote has
+	// enough balance to pay for the fee of our HTLC. If the they cannot
+	// pay the fee if we add another non-dust HTLC, set our available
+	// balance just below the non-dust amount, to avoid attempting such
+	// large HTLCs.
+	case theirBalance < htlcCommitFee && ourBalance >= nonDustHtlcAmt:
+		ourBalance = nonDustHtlcAmt - 1
```

Comment:
I know it is 'correct', but won't it be weird that your balance can go down from a high amount to almost zero if the _other_ party sends you money and brings their balance close to the limit?

Do we need to distinguish between balance (the money that you actually have, including the reserve) and amount sendable?

Reply:
So this method is really used as "amount sendable", as it is called from the link when finding the available bandwidth. It is not used to expose the balance to the user in any way. Maybe changing the method name would make it more clear?

Reply:
Ah yes, that is true. I like `amountSendable` as a name, to make things a bit more clear. But non-blocking.